tests/test_mpdist.py

```diff
@@ -67,9 +67,14 @@ def test_compute_P_ABBA(T_A, T_B):
 def test_mpdist_vect(T_A, T_B):
     m = 3
     ref_mpdist_vect = naive.mpdist_vect(T_A, T_B, m)
+
+    Q_subseq_isconstant = core.rolling_isconstant(T_A, m)
+    T_subseq_isconstant = core.rolling_isconstant(T_B, m)
     μ_Q, σ_Q = naive.compute_mean_std(T_A, m)
     M_T, Σ_T = naive.compute_mean_std(T_B, m)
```

Comment:
I think instead of replacing `core.rolling_isconstant(T_A, m)` with its naive version, we should replace `naive.compute_mean_std` with its performant version. Because, line 69 is using the naive version to calculate `ref_mpdist_vect`. Hence, to compute `comp_mpdist_vect`, we should go with performant functions.

Reply:
Hmm, this is somewhat of a philosophical debate. I tend to prefer the naive version as it should usually be the "most correct" and it would help identify bugs in our performant code when the "most correct" inputs are used to seed our performant functions where possible. In the worst/slowest case, it shouldn't matter because, in theory, the values should be identical. However, we know that the performant version is more likely to have bugs and so this would make it easier/better for tracking down those bugs. 

For example, if we use `core.rolling_isconstant` and `core.compute_mean_std` here and then use their outputs as input to compute `comp_mpdist_vect` and there is an error then we have no idea if the bug is coming from `core.rolling_isconstant`, `core.compute_mean_std`, or `comp_mpdist_vect`. However, if we use `naive.rolling_isconstant` and `naive.compute_mean_std` then, instinctively, I would look at those functions last because, in general, they should be less likely to contribute to the error. Of course they can but it would make it a bit faster for us to isolate where the source of the error is coming from. This is the benefit of having naive implementations of EVERYTHING. We can use them to quickly narrow down the source of the bug/error but at the cost of slightly slower speed (which may or may not be true for small data inputs).

This is a long-winded explanation to try and use naive versions whenever possible. The exception is if:

1. The naive version is exceptionally and painfully slow
2. The performant version is exceptionally simple (less than 5 lines of code) but then the naive version is likely to also be simpler and fast

Reply:
> For example, if we use `core.rolling_isconstant` and `core.compute_mean_std` here and then use their outputs as input to compute `comp_mpdist_vect` and there is an error then we have no idea if the bug is coming from `core.rolling_isconstant`, `core.compute_mean_std`, or `comp_mpdist_vect`. However, if we use `naive.rolling_isconstant` and `naive.compute_mean_std` then, instinctively, I would look at those functions last because, in general, they should be less likely to contribute to the error.

Right! Thank you for the thorough explanation! That makes sense :) Please ignore my other comments.